pkg/controller/plan/migration.go

```diff
@@ -1389,32 +1389,27 @@ func (r *Migration) updateCopyProgress(vm *plan.VMStatus, step *plan.Step) (err
 				var importer *core.Pod
 				var found bool
 				var kErr error
-				if dv.PVC == nil && !r.Plan.IsSourceProviderOCP() {
+
+				if dv.Status.ClaimName == "" {
 					found = false
 				} else {
-					var importerPVC core.PersistentVolumeClaim
-					if r.Plan.IsSourceProviderOCP() {
-						pvc := &core.PersistentVolumeClaim{}
-						err = r.Destination.Client.Get(context.TODO(), types.NamespacedName{
-							Namespace: r.Plan.Spec.TargetNamespace,
-							Name:      dv.Status.ClaimName,
-						}, pvc)
-						if err != nil {
-							log.Error(
-								err,
-								"Could not get PVC for DataVolume.",
-								"vm",
-								vm.String(),
-								"dv",
-								path.Join(dv.Namespace, dv.Name))
-							continue
-						}
-						importerPVC = *pvc
-					} else {
-						importerPVC = *dv.PVC
+					pvc := &core.PersistentVolumeClaim{}
+					err = r.Destination.Client.Get(context.TODO(), types.NamespacedName{
+						Namespace: r.Plan.Spec.TargetNamespace,
+						Name:      dv.Status.ClaimName,
+					}, pvc)
+					if err != nil {
+						log.Error(
+							err,
+							"Could not get PVC for DataVolume.",
+							"vm",
+							vm.String(),
+							"dv",
+							path.Join(dv.Namespace, dv.Name))
+						continue
```

Comment:
I'm ok with having this fix as a temporary & simple solution for backporting, but the core issue here is that we retrieve the DVs using getDVs that return "our struct" but without setting the 'PVC' and so in some places we expect it to be set (when returned from ListVMs) and sometimes it's not.. and in addition, in most of the places we don't even use the two functions of "out struct".. maybe it will be a better idea to drop that struct as @liranr23 wrote above and replace the two methods with utility-methods, so that when "our struct" is returned, we can assume both the DV and the PVC were queried, wdyt?

Reply:
yeah I'm also in favor of dropping it, in this particular case ListVMs doesn't help because the VM doesn't exist yet

Reply:
I'm also in favor of dropping it, and in this particular case ListVMs doesn't help us because there's no VM yet